eth/downloader/downloader.go

```diff
@@ -283,6 +283,15 @@ func (d *Downloader) Synchronising() bool {
 	return atomic.LoadInt32(&d.synchronising) > 0
 }
 
+// SyncBloomContains tests if the syncbloom filter contains the given hash:
+//   - false: the bloom definitely does not contain hash
+//   - true:  the bloom maybe contains hash
+//
+// While the bloom is being initialized (or is closed), all queries will return true.
+func (d *Downloader) SyncBloomContains(hash []byte) bool {
+	return d.stateBloom.Contains(hash)
```

Comment:
stateBloom might be nil though

Reply:
if you are full syncing, or restarted after a fast sync

Reply:
Yeah, the test failures also show that... So either we could do a regular nil-check...._or_ we could make `Contains` return true if the bloom is `nil`... ?

Reply:
```golang
func (b *SyncBloom) Contains(hash []byte) bool {
	bloomTestMeter.Mark(1)
	if b == nil{
		return true
	}
     ...
```

?

Reply:
fixed

Reply:
Now I get it. We have two things that need to care

- state bloom itself is not nil (full sync, or restarted after a fast sync)
- state bloom is not closed (nodes won't be added anymore, in this case state bloom should always say yes)

Reply:
Why not adding the nil check here? 

if stateBloom == nil { return true}